bsi/src/main/java/org/roaringbitmap/bsi/buffer/BitSliceIndexBase.java

```diff
@@ -199,15 +200,21 @@ private ImmutableRoaringBitmap oNeilCompare(BitmapSliceIndex.Operation operation
     for (int i = this.bitCount() - 1; i >= 0; i--) {
       int bit = (predicate >> i) & 1;
       if (bit == 1) {
-        LT = ImmutableRoaringBitmap.or(LT, ImmutableRoaringBitmap.andNot(EQ, this.bA[i]));
+        if (operation == Operation.LT || operation == Operation.LE) {
```

Comment:
Why don't you go a bit further and only call `new MutableRoaringBitmap()` on GT and LT only when needed ? E.g., it seems that LT is only needed when operation is Operation.LT or Operation.LE.

You'd hope that the JIT would see this, but I don't expect so. We forcefully call `new MutableRoaringBitmap()` and eliding the construction is probably too difficult for the optimizer.

Thoughts?

Reply:
Yes, the same idea go in my mind also due to same reason as you - that it could be done by JIT. But I left it over as there are calls outside the method, I suppose it is difficult to check if this does not affect anything other - not necessarily the state of the class, but there can be other effects even not used in this case - network communication, printing to screen etc. Maybe some benchmark would be suitable for this case as there is none in the project or am I overlooked something - may be there is some in other projects?

Reply:
Good remark, I somehow overlooked these useless instantiations.